### PR TITLE
Corrected element hierarchy

### LINKED ISSUE #650: heading heirarchy incorrect
### Description

On the individual profile page there is a h1 followed by an h4. A h1 should be followed by a h2 and size should be controlled by css to make the html semantically correct

### Screenshots

Screenshot 2021-11-09 143337

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by SPARKY892**
resolve #650 

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/660"></a>

### CODE REVIEW

**Review by ssensalo — APPROVED** [review]

**Comment by mohan-murali** [comment]
I personally don't think name and username should be part of heading at all.  A screen reader is going to read it as heading and that is not really the heading of the page. It doesn't give correct meaning.

**Comment by eddiejaoude** [comment]
@mohan-murali this is a page for the user (name) / username, I think it is the only relevant information we have for the page - we would want it specific rather than generic?

**Comment by mohan-murali** [comment]
> @mohan-murali this is a page for the user (name) / username, I think it is the only relevant information we have for the page - we would want it specific rather than generic?

well, then should they both not be in the same heading hierarchy. Having username at a different heading hierarchy seems weird. I don't know, maybe its right. I am just thinking out loud here.

File: src/Components/Profile.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ function Profile({ profile, username }) {
           />
         </Avatar>
         <h1 className="p-m-2">{name}</h1>
```
Comment (krishguptadev) [inline src/Components/Profile.js:28]:
```suggestion
        <h1 className="p-m-2">{name} <span class="p-m-2">({username})</span></h1>
```

File: src/Components/Profile.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ function Profile({ profile, username }) {
...
-        <h4 className="">({username})</h4>
+        <h2 className="p-avatar-username">({username})</h2>
```
Comment (krishguptadev) [inline src/Components/Profile.js:29]:
```suggestion
```

**Review by krishguptadev — CHANGES_REQUESTED** [review]

**Review by krishguptadev — APPROVED** [review]

**Review by Vyvy-vi — COMMENTED** [review]
I think maybe the `name` and the `username` could be in different hierarchies?

**Comment by krishguptadev** [comment]
> I think maybe the `name` and the `username` could be indifferent hierarchies?

I don't think that would be efficient for SEO and yes the issue this one is resolving.

File: src/Components/Profile.js [hunk 3]
```diff
@@ -25,8 +25,7 @@ function Profile({ profile, username }) {
             className="p-mr-2 p-mt-2"
           />
         </Avatar>
-        <h1 className="p-m-2">{name}</h1>
-        <h4 className="">({username})</h4>
+        <h1 className="p-m-2">{name} <span class="p-m-2">({username})</span></h1>
```
Comment (krishguptadev) [inline src/Components/Profile.js:28]:
```suggestion
        <div className="p-m-2 p-d-flex">
          <h1 className="p-m-0 p-r-3">{name}</h1>
          <h2 className="p-avatar-username">({username})</h2>
        </div>
```

**Review by krishguptadev — COMMENTED** [review]
If we wanna implement different level headings for this as @Vyvy-vi suggested.

File: src/Components/Profile.js (hunk 3 above)
Comment (krishguptadev) [reply src/Components/Profile.js:28]: This also fixes the build, not sure about the tests though 😅

Comment (Vyvy-vi) [reply src/Components/Profile.js:28]: moreover, this would also fix the failing test case(`react/no-unknown-property`)

Comment (SPARKY892) [reply src/Components/Profile.js]: This unfortunately makes the username lower than the name
image

File: src/Components/Profile.js [hunk 4]
```diff
@@ -25,8 +25,10 @@ function Profile({ profile, username }) {
             className="p-mr-2 p-mt-2"
           />
         </Avatar>
-        <h1 className="p-m-2">{name}</h1>
-        <h4 className="">({username})</h4>
+        <div className="p-m-2 p-d-flex">
+          <h1 className="p-m-0 p-r-3">{name}</h1>
```
Comment (krishguptadev) [inline src/Components/Profile.js:29]: @SPARKY892 like this?

```suggestion
          <h1 className="p-m-2 p-r-3">{name}</h1>
```

File: src/Components/Profile.css [hunk 5]
```diff
@@ -8,3 +8,7 @@
 .p-avatar-image > img {
   font-size: 0.5em;
 }
+
+.p-avatar-username {
```
Comment (emmalearnscode) [inline src/Components/Profile.css:12]: Custom styles should not be needed.

File: src/Components/Profile.js (hunk 4 above)
Comment (emmalearnscode) [inline src/Components/Profile.js:29]: Using p-mr-2 would give the name a margin only to the right and stop them from being at different levels

File: src/Components/Profile.js [hunk 6]
```diff
@@ -25,8 +25,10 @@ function Profile({ profile, username }) {
...
+          <h2 className="p-avatar-username">({username})</h2>
```
Comment (emmalearnscode) [inline src/Components/Profile.js:30]: Please remove custom css

**Review by emmalearnscode — CHANGES_REQUESTED** [review]
Please try to stick to prime react classes as much as possible. Just a couple of small changes and this will be ready to go :)

File: src/Components/Profile.js [hunk 7]
```diff
@@ -25,8 +25,12 @@ function Profile({ profile, username }) {
             className="p-mr-2 p-mt-2"
           />
         </Avatar>
-        <h1 className="p-m-2">{name}</h1>
-        <h4 className="">({username})</h4>
+        <div className="p-m-2 p-d-flex">
+          <h1 className="p-m-0 p-r-3">{name}</h1>
+          <div className="p-m-2 p-d-flex">
+            <h1 className="p-m-0 p-r-3">{name}</h1>
```
Comment (emmalearnscode) [inline src/Components/Profile.js:31]: There should only ever be one h1 per page due to accessibility reasons. So we either need an h1 and a h2 both name and username should be inside a h1 with an extra span to add different styles to the username.

**Review by emmalearnscode — CHANGES_REQUESTED** [review]
F9037E4C-AD82-4416-84ED-51635EC1DC19

With the current styles these are being offset from each other and it doesn’t look great from a design perspective. Please try to style them so they sit on the same baseline.

File: src/Components/Profile.js (hunk 7 above)
Comment (emmalearnscode) [reply src/Components/Profile.js:31]: How's this coming along? Do you need any help @SPARKY892 ?

**Review by CodingSpecies — COMMENTED** [review]
@SPARKY892 this pr has been open and not active for a while now. Do you need any help? Thank you.

**Review by emmalearnscode — APPROVED** [review]

**Review by krishguptadev — APPROVED** [review]

**Review by Vyvy-vi — APPROVED** [review]
This looks good to me :+1:

**Pull request merged**